### ISSUE TITLE
API changes to support FOM/Keycloak data load

### ISSUE THREAD

**Issue by gormless87** [bcgov/nr-forests-access-management#327]
**Describe the task**
1. Drop client_name column on forest_client table.
2. Revise role_name constraint on fam_role table to include application_id and I think client_id (not sure)
3. Change existing FOM roles to have names "Submitter", "Reviewer".
4. Change role_purpose to nullable on fam_role.

**Acceptance Criteria**
- [ ] first
- [ ] second
- [ ] third

**Additional context**
- Add any other context about the task here.
- Or here

**Comment by ianliuwk1019** [comment]
PR#384 is addressing point 2 (unique constraint - **role_name**, **application_id**), but not the inclusion of client_id. I didn't recall there was conversation about that, also don't feel there is a need to include client_id in the constraint.

Not clear about point 3 means... however, our FOM role records do have "Submitter"
and "Reviewer" as ("FOM_SUBMITTER" and "FOM_REVIEWER"). So we can also cross this out.
@ArogeG

**Comment by ConradBoydElliottGustafson** [comment]
I reviewed PR #340 and PR #384 . Looks good to me.

**Comment by ConradBoydElliottGustafson** [comment]
@ianliuwk1019 Agreed that PR #384 did not need to include client_id in the UK. That would have been a mistake, because it would have allowed two "forest client roles" to have the same name. The point is to have unique names for all the roles underneath a single application.
